test/tasks/before-suite.yaml

```diff
@@ -0,0 +1,107 @@
+# ginkgo-suite is a generic task that executes ginkgo on the given repository and ginkgo suite.
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: before-suite
+  namespace: karpenter-tests
+spec:
+  description: |
+    Execute a karpenter test suite
+  params:
+    - name: cluster-name
+      description: Name of the cluster under test.
+    - name: git-ref
+      description: Git commit, tag, or branch to check out. Requires a corresponding Karpenter snapshot release.
+  results:
+    - name: account-id
+    - name: region
+  workspaces:
+    - name: ws
+  steps:
+  - name: detect-environment
+    image: 767520670908.dkr.ecr.us-west-2.amazonaws.com/karpenter-test-infrastructure:v4
+    workingDir: $(workspaces.ws.path)
+    script: |
+      TOKEN=$(curl -X PUT "http://169.254.169.254/latest/api/token" -H "X-aws-ec2-metadata-token-ttl-seconds: 21600")
+      curl -H "X-aws-ec2-metadata-token: $TOKEN" -v http://169.254.169.254/latest/meta-data/identity-credentials/ec2/info | grep "AccountId" | cut -f4 -d"\"" > $(results.account-id.path)
+      curl -H "X-aws-ec2-metadata-token: $TOKEN" -v http://169.254.169.254/latest/meta-data/placement/region > $(results.region.path)
+
+  - name: checkout-repository
+    image: alpine/git
+    workingDir: $(workspaces.ws.path)
+    script: |
+      git clone https://github.com/aws/karpenter.git .
+      git checkout $(params.git-ref)
+
+  - name: create-iam-policies
+    image: 767520670908.dkr.ecr.us-west-2.amazonaws.com/karpenter-test-infrastructure:v4
+    workingDir: $(workspaces.ws.path)
+    script: |
+      aws cloudformation deploy \
+        --stack-name KarpenterIAM-$(params.cluster-name) \
+        --template-file website/content/en/preview/getting-started/getting-started-with-eksctl/cloudformation.yaml \
+        --capabilities CAPABILITY_NAMED_IAM \
+        --parameter-overrides "ClusterName=$(params.cluster-name)"
+
+  - name: create-cluster
+    image: 767520670908.dkr.ecr.us-west-2.amazonaws.com/karpenter-test-infrastructure:v4
+    workingDir: $(workspaces.ws.path)
+    script: |
+      cmd="create"
+      eksctl get cluster --name $(params.cluster-name) && cmd="upgrade"
+      eksctl ${cmd} cluster -f - <<EOF
+      ---
+      apiVersion: eksctl.io/v1alpha5
+      kind: ClusterConfig
+      metadata:
+        name: $(params.cluster-name)
+        region: $(cat $(results.region.path))
+        version: "1.22"
+        tags:
+          karpenter.sh/discovery: $(params.cluster-name)
+      managedNodeGroups:
+        - instanceType: m5.large
+          amiFamily: AmazonLinux2
+          name: $(params.cluster-name)-system-pool
+          desiredCapacity: 2
+          minSize: 2
+          maxSize: 2
+          taints:
+          - key: CriticalAddonsOnly
+            value: "true"
+            effect: NoSchedule
+      iam:
+        withOIDC: true
+      EOF
+
+      eksctl create iamidentitymapping \
```

Comment:
nit: Can we put this and the following command in another step? Something like `add-authentication`, since it's not actually creating a cluster.

Reply:
I originally had this broken out into single command steps, but this ends up being really verbose in tekton. I'm not sure it's materially different, since it's all part of the core cluster configuration (and eksctl even handles the cascading deletes)

Reply:
sgtm